api_tests/crossref/views/test_crossref_email_response.py

```diff
@@ -0,0 +1,160 @@
+import mock
+import pytest
+import hmac
+import hashlib
+
+from osf_tests import factories
+from website import settings
+
+
+@pytest.mark.django_db
+class TestCrossRefEmailResponse:
+
+    def create_mailgun_request_context(self, response_data):
+        if not settings.MAILGUN_API_KEY:
```

Comment:
Rather than require this to be set, you can override it. pytest-django allows you to temporarily override a setting: https://pytest-django.readthedocs.io/en/latest/helpers.html#settings

Reply:
One way to make this developer-friendly would be to make a fixture that returns a function. That way you could use the `settings` fixture mentioned above, and you could provide parameters for changing the body and 

```python
@pytest.fixture()
def make_mailgun_payload(settings):
    # temporarily override MAILGUN_API_KEY
    settings.MAILGUN_API_KEY = 'notsosecret'
    def func(body, signature=None):
        return {
            'X-Mailgun-Sscore': ....,
             'body-plain': body.strip(),
            'signature': signature or hmac.new(....),
            ...
        }
    return func
```

Then you'd use it like so:

```python
def test_...(self, make_mailgun_payload, success_xml):
    ...
    payload = make_mailgun_payload(success_xml)
    app.post(url, payload)
    ...
```